purchases/src/main/kotlin/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -64,6 +65,9 @@ import kotlin.time.Duration
 private const val RECONNECT_TIMER_START_MILLISECONDS = 1L * 1000L
 private const val RECONNECT_TIMER_MAX_TIME_MILLISECONDS = 1000L * 60L * 15L // 15 minutes
 
+// Arbitrary. There's an IllegalStateException on Samsung devices that try to bind to the service more than 999 times
```

Comment:
:+1

Reply:
It looks like this is Samsung specific and Android 12 and 13, so most likely a bug on their OS

For what I've read, each bind increments the counter, so if there's a bind hanging and 999+ transfers have been requested, it's possible this exception gets thrown. The binding happens on every startConnection.

Reply:
This is a good read by the way

https://stackoverflow.com/questions/76512231/do-samsung-phones-running-android-13-have-an-intent-bug